spec/clang_spec.lua

```diff
@@ -178,4 +177,154 @@ describe("cursor kind", function()
                 end)
                 assert.are.same(expected, children)     
         end)
-end)
\ No newline at end of file
+end)
+
+function find_decl(cursor)
+        local decl_cursor
+        cursor:visitChildren(function(cursor, parent)
+                decl_cursor = cursor
+                return "continue"
+        end)
+        return decl_cursor
+end
+
+describe("type object creation", function() 
+        it("succeeds in creating type object", function() 
+                local parser = luaclang.newParser("spec/function.c")
+                local cursor = parser:getCursor()   
+                local cursor_type = cursor:getType()             
+                assert.are.same('userdata', type(cursor_type))
+        end)       
+end)
+
+describe("type spelling", function() 
+        it("succeeds in matching the expected type spelling", function()
+                local parser = luaclang.newParser("spec/function.c")
+                local cursor = parser:getCursor()
+                cursor = find_decl(cursor)
+                local cursor_type = cursor:getType()
+                local cursor_type_spelling = cursor_type:getTypeSpelling()
+                assert.are.equals(cursor_type_spelling, "void (float, float *)")
```

Comment:
```suggestion
                assert.are.equals("void (float, float *)", cursor_type_spelling)
```
Invert the arguments, the expected one should come first. Please fix this in the whole PR.